keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -291,6 +292,12 @@ open class SqlResourceRepository(
         .where(DIFF_FINGERPRINT.ENTITY_ID.eq(id))
         .execute()
     }
+    sqlRetry.withRetry(WRITE) {
+      jooq.deleteFrom(PAUSED)
```

Comment:
This was previously missing, and now enforces the assumption I mentioned in the PR description that, when a resource is deleted, a matching pause record should also be deleted as the resource effectively stops being managed.